cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1507,11 +1514,46 @@ protected String getServiceAccountName() {
         return varList;
     }
 
+    /**
+     * Validates provided OAuth configuration. Throws InvalidResourceException when OAuth configuration contains forbidden combinations.
+     *
+     * @param oAuth     OAuth type authentication object
+     */
+    @SuppressWarnings("checkstyle:BooleanExpressionComplexity")
+    private static void validateOauth(KafkaListenerAuthenticationOAuth oAuth) {
+        if (oAuth.getIntrospectionEndpointUri() == null && oAuth.getJwksEndpointUri() == null) {
+            log.error("Introspection endpoint URI or JWKS endpoint URI has to be specified");
+            throw new InvalidResourceException("Introspection endpoint URI or JWKS endpoint URI has to be specified");
+        }
+
+        if (oAuth.getValidIssuerUri() == null) {
+            log.error("Valid Issue URI has to be specified");
+            throw new InvalidResourceException("Valid Issue URI has to be specified");
+        }
+
+        if ((oAuth.getJwksExpirySeconds() > 0 && oAuth.getJwksRefreshSeconds() > 0 && oAuth.getJwksExpirySeconds() < oAuth.getJwksRefreshSeconds() + 60) ||
```

Comment:
I would do this validation at the end only - if they are using refresh and expiration configs with introspection endpoint it is better to tell them that error first than this one.